src/core/environment_test.go

```diff
@@ -492,6 +493,152 @@ func TestWriteFile(t *testing.T) {
 	assert.NoFileExists(t, file.GetName())
 }
 
+func getUbuntuOsReleaseInfo() string {
+	return "NAME=\"Ubuntu\"\n" +
+	"VERSION=\"20.04.5 LTS (Focal Fossa)\"\n" +
+	"VERSION_ID=\"20.04\"\n" +
+	"ID=ubuntu\n" +
+	"ID_LIKE=debian\n" +
+	"PRETTY_NAME=\"Ubuntu 20.04.5 LTS\"\n" +
+	"HOME_URL=\"https://www.ubuntu.com/\"\n" +
+	"SUPPORT_URL=\"https://help.ubuntu.com/\"\n" +
+	"BUG_REPORT_URL=\"https://bugs.launchpad.net/ubuntu/\"\n" +
+	"PRIVACY_POLICY_URL=\"https://www.ubuntu.com/legal/terms-and-policies/privacy-policy\"\n" +
+	"VERSION_CODENAME=focal\n" +
+	"UBUNTU_CODENAME=focal"
+}
+
+func getFedoraOsReleaseInfo() string {
+	return "NAME=Fedora\n" +
+	"VERSION=\"32 (Workstation Edition)\"\n" +
+	"ID=fedora\n" +
+	"VERSION_ID=32\n" +
+	"PRETTY_NAME=\"Fedora 32 (Workstation Edition)\"\n" +
+	"ANSI_COLOR=\"0;38;2;60;110;180\"\n" +
+	"LOGO=fedora-logo-icon\n" +
+	"CPE_NAME=\"cpe:/o:fedoraproject:fedora:32\"\n" +
+	"HOME_URL=\"https://fedoraproject.org/\"\n" +
+	"DOCUMENTATION_URL=\"https://docs.fedoraproject.org/en-US/fedora/f32/system-administrators-guide/\"\n" +
+	"SUPPORT_URL=\"https://fedoraproject.org/wiki/Communicating_and_getting_help\"\n" +
+	"BUG_REPORT_URL=\"https://bugzilla.redhat.com/\"\n" +
+	"REDHAT_BUGZILLA_PRODUCT=\"Fedora" +
+	"REDHAT_BUGZILLA_PRODUCT_VERSION=32\n" +
+	"REDHAT_SUPPORT_PRODUCT=\"Fedora" +
+	"REDHAT_SUPPORT_PRODUCT_VERSION=32\n" +
+	"PRIVACY_POLICY_URL=\"https://fedoraproject.org/wiki/Legal:PrivacyPolicy\"\n" +
+	"VARIANT=\"Workstation Edition\"\n" +
+	"VARIANT_ID=workstation"
+}
+
+func TestParseOsReleaseFile(t *testing.T) {
+	tests := []struct {
+		name string
+		osReleaseContent string
+		expect map[string]string
+	} {
+		{
+			name: "TestUbuntuOsReleaseInfo",
+			osReleaseContent: getUbuntuOsReleaseInfo(),
+			expect: map[string]string{
+				"VERSION_ID": "20.04",
+				"VERSION": "20.04.5 LTS (Focal Fossa)",
+				"VERSION_CODENAME": "focal",
+				"NAME": "Ubuntu",
+				"ID": "ubuntu",
+			},
+		},
+		{
+			name: "TestFedoraOsReleaseInfo",
+			osReleaseContent: getFedoraOsReleaseInfo(),
+			expect: map[string]string{
+				"VERSION_ID": "32",
+				"VERSION": "32 (Workstation Edition)",
+				"VERSION_CODENAME": "",
+				"NAME": "Fedora",
+				"ID": "fedora",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reader := strings.NewReader(tt.osReleaseContent)
+			osRelease, _ := parseOsReleaseFile(reader)
+			assert.Equal(t, tt.expect["VERSION_ID"] , osRelease["VERSION_ID"])
+			assert.Equal(t, tt.expect["VERSION"] , osRelease["VERSION"])
+			assert.Equal(t, tt.expect["VERSION_CODENAME"] , osRelease["VERSION_CODENAME"])
+			assert.Equal(t, tt.expect["NAME"] , osRelease["NAME"])
+			assert.Equal(t, tt.expect["ID"] , osRelease["ID"])
```

Comment:
testify's `Equal` boils down to [`reflect.DeepEqual`](https://pkg.go.dev/reflect#DeepEqual), which supports maps. We only need one assert to get good results:

```suggestion
			assert.Equal(t, tt.expect , osRelease)
```

For example, given this test:

```go
func TestFoo(t *testing.T) {
	expected := map[string]int{"a": 1, "b": 2}
	actual := map[string]int{"b": 2, "c": 3}
	assert.Equal(t, expected, actual)
}
```

`testify` prints output that tells us exactly what was different:

```text
--- FAIL: TestFoo (0.00s)
    /path/to/test.go:32: 
        	Error Trace:	/path/to/test.go:32
        	Error:      	Not equal: 
        	            	expected: map[string]int{"a":1, "b":2}
        	            	actual  : map[string]int{"b":2, "c":3}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]int) (len=2) {
        	            	- (string) (len=1) "a": (int) 1,
        	            	- (string) (len=1) "b": (int) 2
        	            	+ (string) (len=1) "b": (int) 2,
        	            	+ (string) (len=1) "c": (int) 3
        	            	 }
        	Test:       	TestFoo
FAIL
```

Reply:
This was an interesting read. Thanks. I will make fix.